File: selenium/src/java/org/openqa/selenium/internal/XPathLookupStrategy.java
package org.openqa.selenium.internal;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;

public class XPathLookupStrategy implements LookupStrategy {
    public WebElement find(WebDriver driver, String use) {
        if (use.endsWith("/"))
            use = use.substring(0, use.length() - 1);
        return driver.findElement(By.xpath(use));
    }
}

File: selenium/src/csharp/WebdriverBackedSelenium/Internal/SeleniumEmulation/LookupStrategies/XPathLookupStrategy.cs
using System;
using OpenQA.Selenium;

namespace Selenium
{
    internal class XPathLookupStrategy : ILookupStrategy
    {
        public IWebElement Find(IWebDriver driver, string use)
        {
            return driver.FindElement(By.XPath(use));
        }
    }
}
